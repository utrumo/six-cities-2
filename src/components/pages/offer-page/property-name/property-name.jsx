import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import {connect} from 'react-redux';
import {OfferTypeToPresentName} from 'shared/const';
import {DataSelector, DataOperation} from 'store';

class PropertyName extends PureComponent {
  constructor(props) {
    super(props);

    this._handleButtonClick = this._handleButtonClick.bind(this);
  }

  _handleButtonClick() {
    const {id, isFavorite, onButtonClick} = this.props;
    onButtonClick(id, isFavorite);
  }

  render() {
    const {title, type, isFavorite} = this.props;
    return (
      <div className="property__name-wrapper">
        <h1 className="property__name">{title}</h1>
        <p
          style={{padding: `5px 0 15px`, fontSize: `16px`, textAlign: `center`}}
          className="place-card__type"
        >
          {OfferTypeToPresentName[type]}
        </p>
        <button
          onClick={this._handleButtonClick}
          className={classNames(
              `property__bookmark-button`,
              {'property__bookmark-button--active': isFavorite},
              `button`,
          )}
          type="button"
        >
          <svg className="property__bookmark-icon place-card__bookmark-icon" width="31" height="33">
            <use xlinkHref="#icon-bookmark"></use>
          </svg>
          <span className="visually-hidden">To bookmarks</span>
        </button>
      </div>
    );
  }
}


PropertyName.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired,
  id: PropTypes.number.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  onButtonClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  title: DataSelector.getCurrentOfferTitle(state),
  type: DataSelector.getCurrentOfferType(state),
  id: DataSelector.getCurrentOfferId(state),
  isFavorite: DataSelector.getCurrentOfferIsFavoriteFlag(state),
});

const mapDispatchToProps = {
  onButtonClick: DataOperation.toggleFavoriteStatus,
};

export {PropertyName};
export default connect(mapStateToProps, mapDispatchToProps)(PropertyName);
