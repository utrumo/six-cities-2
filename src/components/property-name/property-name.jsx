import React from 'react';
import PropTypes from 'prop-types';
import {OfferTypeToPresentName} from '../../shared/const';

const PropertyName = (props) => {
  const {title, type} = props;
  return (
    <div className="property__name-wrapper">
      <h1 className="property__name">{title}</h1>
      <p
        style={{padding: `5px 0 15px`, fontSize: `16px`, textAlign: `center`}}
        className="place-card__type"
      >
        {OfferTypeToPresentName[type]}
      </p>
      <button className="property__bookmark-button button" type="button">
        <svg className="property__bookmark-icon" width="31" height="33">
          <use xlinkHref="#icon-bookmark"></use>
        </svg>
        <span className="visually-hidden">To bookmarks</span>
      </button>
    </div>
  );
};

PropertyName.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf([`apartment`, `room`, `house`, `hotel`]).isRequired
};

export default PropertyName;
