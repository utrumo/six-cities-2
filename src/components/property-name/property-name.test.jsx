import React from 'react';
import renderer from 'react-test-renderer';
import {PropertyName} from './property-name.jsx';

it(`PropertyName should renders correct.`, () => {
  const tree = renderer.create(<PropertyName title="Test title" type="room"/>).toJSON();
  expect(tree).toMatchSnapshot();
});
