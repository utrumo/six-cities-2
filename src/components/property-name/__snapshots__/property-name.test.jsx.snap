// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyName should renders correct. 1`] = `
<div
  className="property__name-wrapper"
>
  <h1
    className="property__name"
  >
    Test title
  </h1>
  <p
    className="place-card__type"
    style={
      Object {
        "fontSize": "16px",
        "padding": "5px 0 15px",
        "textAlign": "center",
      }
    }
  >
    Private Room
  </p>
  <button
    className="property__bookmark-button property__bookmark-button--active button"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="property__bookmark-icon place-card__bookmark-icon"
      height="33"
      width="31"
    >
      <use
        xlinkHref="#icon-bookmark"
      />
    </svg>
    <span
      className="visually-hidden"
    >
      To bookmarks
    </span>
  </button>
</div>
`;
